src/ui/components/filters/facetscomponent.js

```diff
@@ -237,14 +260,17 @@ export default class FacetsComponent extends Component {
       this._filterbox.remove();
     }
 
-    let { filters, resultsContext } = this._state.get();
+    let filters = this._getFacets();
```

Comment:
could we do this transformation in setState()? I don't think there's a huge difference but it seems a little more thematic of the SDK, also a few less this._state.get() calls and we probably don't need to deepclone. It feels like transformFacets is a transformData but with a stricter, less leaky promise

Reply:
Are you saying to call transformFacets during setState and store that in the state?

Reply:
yeah exactly that, instead of the original "data". we can also use the "isNoResults" state

Reply:
updated to do all the transforming in setState!